pkgs/by-name/tr/tre/package.nix

```
@@ -1,30 +1,57 @@
 {
   lib,
   stdenv,
-  fetchurl,
-  fetchpatch,
+  fetchFromGitHub,
+  autoreconfHook,
+  automake,
+  darwin,
+  gettext,
+  libiconv,
+  libtool,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "tre";
-  version = "0.8.0";
-  src = fetchurl {
-    url = "https://laurikari.net/tre/${pname}-${version}.tar.gz";
-    sha256 = "1pd7qsa7vc3ybdc6h2gr4pm9inypjysf92kab945gg4qa6jp11my";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "laurikari";
+    repo = "tre";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-5O8yqzv+SR8x0X7GtC2Pjo94gp0799M2Va8wJ4EKyf8=";
   };
 
-  patches = [
-    (fetchpatch {
-      url = "https://sources.debian.net/data/main/t/tre/0.8.0-6/debian/patches/03-cve-2016-8859";
-      sha256 = "0navhizym6qxd4gngrsslbij8x9r3s67p1jzzhvsnq6ky49j7w3p";
-    })
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    automake
+    libtool
   ];
 
+  buildInputs =
+    [
+      gettext
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      libiconv
+    ];
+
+  autoreconfPhase = ''
+    runHook preAutoreconf
+    ./utils/autogen.sh
+    runHook postAutoreconf
+  '';
```

```suggestion
  preConfigure = ''
    ./utils/autogen.sh
  '';
```
autoreconfHook would have done `appendToVar preConfigurePhases autoreconfPhase`. Simplify to just use preConfigure directly.

---

This breaks the build. `autoconf` must be present since the build fails as `autom4te` is not present when building with this suggested change. I won't accept this suggestion since it breaks building.

---

Then put autoconf in nativeBuildInputs

---

And automake too, if it is also required.

---

Fixed in latest commit.